### PR TITLE
Make the signature of fill_out consistent with fill_.

### PR DESCRIPTION
**Pull request by xuhdev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22761 Make the signature of fill_out consistent with fill_.**
* #22760 Move the body of fill_kernel_impl into fill_kernel_cuda
* #22758 Move fill and fill_diagonal to Fill.cpp, Fill.h, and FillKernel.{cpp,cu}

value should be const.

Differential Revision: [D16257779](https://our.internmc.facebook.com/intern/diff/D16257779)

### CODE REVIEW

File: aten/src/ATen/native/Fill.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@ Tensor& fill_out(Tensor& self, const Scalar value) {
   return self;
 }
 
-Tensor& fill_(Tensor& self, Scalar value) {
+Tensor& fill_(Tensor& self, const Scalar value) {
```
Comment (colesbury) [inline aten/src/ATen/native/Fill.cpp]: We don't usually label value types const

Comment (colesbury) [reply aten/src/ATen/native/Fill.cpp]: (but we're not very consistent about it)

Comment (xuhdev) [reply aten/src/ATen/native/Fill.cpp]: I changed this to consistently not use const

**Review by colesbury — APPROVED** [review]

**Pull request closed without merging**